### PR TITLE
Add some cross references in the docs, move an example to a dartpad example

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This adds some "See also" links to some docs for `TweenAnimationBuilder` and `ValueListenableBuilder`.

Also, moved a "snippet" example in `ValueListenableBuilder` into the examples directory as a Dartpad example.

## Tests
 - Added test for the example.

### CODE REVIEW

File: examples/api/lib/widgets/value_listenable_builder/value_listenable_builder.0.dart [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+class ValueListenableBuilderExample extends StatefulWidget {
+  const ValueListenableBuilderExample({super.key});
+
+  @override
+  State<ValueListenableBuilderExample> createState() => _ValueListenableBuilderExampleState();
+}
+
+class _ValueListenableBuilderExampleState extends State<ValueListenableBuilderExample> {
+  final ValueNotifier<int> _counter = ValueNotifier<int>(0);
+  final Widget goodJob = const Text('Good job!');
```
Comment (goderbauer) [inline examples/api/lib/widgets/value_listenable_builder/value_listenable_builder.0.dart:31]: nit: this could be private as well.

Comment (goderbauer) [reply examples/api/lib/widgets/value_listenable_builder/value_listenable_builder.0.dart:31]: Actually, this seems unused; so should probably just be removed?

**Review by goderbauer — APPROVED** [review]
LGTM

File: examples/api/lib/widgets/value_listenable_builder/value_listenable_builder.0.dart (hunk 1 above)
Comment (gspencergoog) [reply examples/api/lib/widgets/value_listenable_builder/value_listenable_builder.0.dart]: Yeah, left over from the code I started from.  Removed.

**Pull request merged**